thin-replica-server/include/thin-replica-server/thin_replica_impl.hpp

```diff
@@ -728,10 +736,11 @@ class ThinReplicaImpl {
                               ServerWriterT* stream,
                               kvbc::BlockId start,
                               kvbc::BlockId end,
-                              std::shared_ptr<kvbc::KvbAppFilter> kvb_filter) {
+                              std::shared_ptr<kvbc::KvbAppFilter> kvb_filter,
+                              ThinReplicaServerMetrics& metrics) {
     boost::lockfree::spsc_queue<kvbc::KvbFilteredUpdate> queue{10};
     std::atomic_bool close_stream = false;
-
+    uint16_t update_aggregator_counter = 0;
```

Comment:
Why are we creating a new counter here? We already have a counter defined [here](https://github.com/vmware/concord-bft/blob/72cfb4761fd6a7236b5a9a21b354c1c4f4775ce4/thin-replica-server/include/thin-replica-server/thin_replica_impl.hpp#L287). We must use a single counter for a single component (TRS) to track the frequency with which we want to update the metrics aggregator. The idea is to update the metrics after every 100 blocks, whether we get them from storage or live update queue.

Reply:
With this change we will have two counters to identify when to update aggregator. Is that the purpose? We use a separate counter for storage updates and live updates?
Consider the scenario, TRS streams 70 updates from storage, and then 30 updates from live update queue. In this case we will not update the aggregator until 100 updates are sent from the live update queue, no? Am I missing anything? For such scenarios, I would suggest we use the counter declared [here](https://github.com/vmware/concord-bft/blob/3b66c30c521980f85f39058c2cdbb64a960b50d2/thin-replica-server/include/thin-replica-server/thin_replica_impl.hpp#L292), pass it to `readFromKvbAndSendData` via `syncAndSend`. WDYT?

Reply:
Addressed but in different way. Thanks.

Reply:
Thanks for the fix.